src/comments.js

```diff
@@ -734,6 +735,19 @@ function handleClassMethodComments(enclosingNode, comment) {
   return false;
 }
 
+function handleTypeAliasComments(enclosingNode, followingNode, comment) {
+  if (
+    enclosingNode &&
+    enclosingNode.type === "TypeAlias" &&
+    followingNode &&
+    followingNode.type === "ObjectTypeAnnotation"
```

Comment:
It's likely going to be an issue with all the types, not only objects.

What about if `enclosingNode === "TypeAlias"` then put the comment as leading of the enclosingNode?